crates/rpc/rpc-eth-api/src/helpers/state.rs

```diff
@@ -48,13 +50,16 @@ pub trait EthState: LoadState + SpawnBlocking {
         address: Address,
         block_id: Option<BlockId>,
     ) -> impl Future<Output = Result<U256, Self::Error>> + Send {
-        self.spawn_blocking_io(move |this| {
-            Ok(this
-                .state_at_block_id_or_latest(block_id)?
-                .account_balance(&address)
-                .map_err(Self::Error::from_eth_err)?
-                .unwrap_or_default())
-        })
+        async move {
+            let state = self.state_at_block_id_or_latest(block_id).await?;
+            self.spawn_blocking_io(move |_this| {
```

Comment:
this is a bit tricky, because for the pending state we now need to mix async and IO.

however, this is already what

https://github.com/paradigmxyz/reth/blob/d53ecc5ab0b2931e0fe6cdc906eff5c361767c50/crates/rpc/rpc-eth-api/src/helpers/blocking_task.rs#L44-L61

does so we could add `spawn_blocking_fut` or something that accepts a future instead, in which we could await the `state_at_block_id_or_latest`

which would be equivalent to what we already do now